velox/functions/prestosql/ArrayIntersectExcept.cpp

```diff
@@ -269,29 +271,143 @@ class ArrayIntersectExceptFunction : public exec::VectorFunction {
 
   // If there's a `constantSet`, whether it refers to left or right-hand side.
   const bool isLeftConstant_{false};
-};
+}; // class ArrayIntersectExcept
+
+template <typename T>
+class ArraysOverlapFunction : public exec::VectorFunction {
```

Comment:
Does this function support all types or only scalar types for the array elements? If it only supports scalar types, please, update the documentation and PR description to clarify this limitation. How are you thinking about adding support for all types?

Reply:
For now this only supports scalar types, similar to Array intersect and except functions. We have not thought about complex types support yet.